src/server/scripts/Spells/spell_generic.cpp

```diff
@@ -1668,10 +1668,15 @@ class spell_steal_essence_visual : public AuraScript
 
     void HandleRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
     {
-        if (!GetCaster() || !GetCaster()->GetOwner())
+        Unit* caster = GetCaster();
+        if (!caster)
             return;
 
         Creature* pet = GetCaster()->ToCreature();
+        Unit* owner = caster->GetOwner();
+        if (!pet || !owner)
+            return;
+
         pet->CastSpell(GetCaster()->GetOwner(), SPELL_CREATE_TOKEN);
```

Comment:
Why don't you use here `owner` instead of calling `GetCaster()->GetOwner()` again?

Reply:
Fair point, even if it is outside of the added code.

Reply:
Fixed. Thanks!